forms/models/elements/subform.js

```diff
@@ -256,7 +256,14 @@ define(function (require) {
 
       return new Promise(function (resolve, reject) {
         Promise.all(promises).then(function (values) {
-          resolve(values);
+          values = _.reduce(values, function (memo, value) {
+            if (value._action === 'remove' && (value.id === null || value.id === undefined)) {
```

Comment:
> ensure that value.id is either a non-empty String or a Number.

Can the id ever be 0? I originally had `!value.id` but that would of removed anything that has an id of 0, which is why i went more specific. if you follow the code the only way that the id field could be empty is if its not set by us, and that test is just `else if (model.attributes.id) {`, which fails when id is set to 0. 

I know MySQL has autoincrement field starting at 1, but im not sure where this data goes into a db we control or is just passed through ??

I've beefed up the tests to check for every type of id (undefined, null, number, string, NaN, but im not sure what the correct behavior setting to id to 0 should be.